File: Data Structures/BinaryTrees/CheckBST.py
class Node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

def isBST(root):
    return isBSTUtil(root, INT_MIN, INT_MAX)

def isBSTUtil(node, low, high):
    if node is None:
        return True
    if node.data < low or node.data > high:
        return False
    return isBSTUtil(node.left, low, node.data - 1) and isBSTUtil(node.right, node.data + 1, high)

File: Data Structures/BinaryTrees/CheckBST.java
class CheckBST
{
    static class Node
    {
        int data;
        Node left, right;
        public Node(int data)
        {
            this.data = data;
            this.left = this.right = null;
        }
    }

    static boolean isBST(Node root)
    {
        return isBSTUtil(root, Integer.MIN_VALUE, Integer.MAX_VALUE);
    }

    static boolean isBSTUtil(Node node, int min, int max)
    {
        if(node == null)
            return true;
        if(node.data < min || node.data > max)
            return false;
        return (isBSTUtil(node.left, min, node.data - 1) && isBSTUtil(node.right, node.data + 1, max));
    }
}

